From 6c33fb58869ffb17106047c45ab8d3856966eaf7 Mon Sep 17 00:00:00 2001
From: Anuj Mittal <anuj.mittal@intel.com>
Date: Tue, 26 Mar 2019 14:11:29 +0800
Subject: [PATCH] point to correct clang project and tblgen

Point to correct path for clang project as per the way we unpack. Also
let llvm-tblgen path be passed from recipe itself.

Also since we're going to do the patching ourselves, no need to look for
git through cmake.

Upstream-Status: Inappropriate [OE specific]
---
 CMakeLists.txt | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 174133b..c769f08 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -53,7 +53,7 @@ endif(CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR)
 include(AddLLVM)
 include(TableGen)
 
-find_package(Git REQUIRED)
+#find_package(Git REQUIRED)
 
 if (NOT WIN32)
     add_subdirectory( linux_linker )
@@ -80,7 +80,7 @@ set(TARGET_NAME ${COMMON_CLANG_LIBRARY_NAME}${BUILD_PLATFORM} )
 
 if(NOT USE_PREBUILT_LLVM)
     set(TARGET_BRANCH "ocl-open-80")
-    set(CLANG_SOURCE_DIR ${LLVM_SOURCE_DIR}/tools/clang)
+    set(CLANG_SOURCE_DIR ${LLVM_SOURCE_DIR}/../clang)
     set(CLANG_BASE_REVISION a03da8be08a208122e292016cb6cea1f30229677)
 
     set(SPIRV_SOURCE_DIR ${LLVM_SOURCE_DIR}/projects/llvm-spirv)
@@ -102,7 +102,7 @@ endif(NOT USE_PREBUILT_LLVM)
 set (COMPILE_OPTIONS_TD  opencl_clang_options.td)
 set (COMPILE_OPTIONS_INC opencl_clang_options.inc)
 
-set(LLVM_TABLEGEN_EXE "llvm-tblgen")
+#set(LLVM_TABLEGEN_EXE "llvm-tblgen")
 set(LLVM_TARGET_DEFINITIONS ${COMPILE_OPTIONS_TD})
 if(USE_PREBUILT_LLVM)
   set(TABLEGEN_ADDITIONAL -I ${LLVM_INCLUDE_DIRS})
@@ -153,7 +153,7 @@ endif()
 
 if(NOT USE_PREBUILT_LLVM)
     set(CLANG_BINARY_DIR ${LLVM_BINARY_DIR}/tools/clang/)
-    set(CLANG_SOURCE_DIR ${LLVM_MAIN_SRC_DIR}/tools/clang/)
+    set(CLANG_SOURCE_DIR ${LLVM_MAIN_SRC_DIR}/../clang/)
     include_directories(
         ${CLANG_BINARY_DIR}/include # for tablegened includes
         ${CLANG_SOURCE_DIR}/include # for basic headers
-- 
2.19.1

